bin/init_exercise.py

```diff
@@ -0,0 +1,411 @@
+    item = description.lower().replace(' ', '_')
+    item = [c for c in item if c in ITEM_NAME_CHARS]
+    while len(item) > 0 and item[0] in string.digits:
+        item = item[1:]
+    if len(item) == 0:
+        raise ValueError("Could not produce an item name from " + description)
+    return item
+
+
+def url_for(name, file):
+    return "https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/{name}/{file}".format(
+        name=name, file=file,
+    )
+
+
+def get_problem_specification(name):
+    """
+    Try to get problem specifications for the exercise of the given name.
+
+    If the problem specifications repo doesn't exist, returns None.
+    Otherwise, returns a dict, of which the values might be None or str.
+    """
+    try:
+        with urllib.request.urlopen(url_for(name, 'canonical-data.json')) as response:
+            return json.loads(response.read())
+    except (urllib.request.URLError, json.JSONDecodeError):
+        pass
+
+
+@contextmanager
+def inside(path):
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+
+
+def make_exercise(name):
+    "Make a new exercise with the specified name"
+    with inside(EXERCISES):
+        if os.path.exists(name):
+            print("{} already exists; aborting".format(name), file=sys.stderr)
+            sys.exit(1)
+        subprocess.run(['cargo', 'new', name])
+    exercise_dir = os.path.join(EXERCISES, name)
+    # blank out the default lib.rs
+    with inside(exercise_dir):
+        with open(os.path.join('src', 'lib.rs'), 'w') as lib_rs:
+            lib_rs.write('')
+        os.mkdir('tests')
+        with inside('tests'):
+            with open('{}.rs'.format(name), 'w') as tests_rs:
+                print("use {}::*;".format(name), file=tests_rs)
+                print(file=tests_rs)
+
+    cd = get_problem_specification(name)
+    if cd is None:
+        print("No problem specification for {} found".format(name))
+        make_new_exercise(name, exercise_dir)
+        with inside(exercise_dir):
+            generate_readme(name, False)
+    else:
+        make_exercise_with_specifications(name, exercise_dir, cd)
+        with inside(exercise_dir):
+            generate_readme(name, True)
+
+
+def make_new_exercise(name, exercise_dir):
+    print("Creating new exercise from scratch...")
+    with inside(exercise_dir):
+        os.mkdir('.meta')
+        with inside('.meta'):
+            with open('description.md', 'w') as description:
+                print("Describe your exercise here", file=description)
+                print(file=description)
+                print("Don't forget that `README.md` is automatically generated; update this within `.meta\description.md`.", file=description)
+            with open('metadata.yml', 'w') as metadata:
+                print("---", file=metadata)
+                print("blurb: \"{} (created {})\"".format(

```

Comment:
Putting the "created" portion here may encourage contributors to leave it there.

I think we do not want to encourage that, because it doesn't seem to be relevant to http://exercism.io/languages/rust/exercises when the exercise was created. I would remove it.